File: frontend/apps/marveline/src/types/notification.ts
export interface Notification {
  id: number
  tenant_id: number
  user_id: number | null
  type: string
  title: string
  message: string | null
  link: string | null
  is_read: boolean
  created_at: string
  read_at: string | null
}

export interface NotificationList {
  items: Notification[]
  total: number
  unread_count: number
}

File: app/schemas/notification.py
"""Schémas Pydantic — Notifications."""
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class NotificationRead(BaseModel):
    id: int
    tenant_id: int
    user_id: Optional[int]
    type: str
    title: str
    message: Optional[str]
    link: Optional[str]
    is_read: bool
    created_at: datetime
    read_at: Optional[datetime]

    model_config = {"from_attributes": True}


class NotificationList(BaseModel):
    items: list[NotificationRead]
    total: int
    skip: int
    limit: int
    unread_count: int
